llama_index/storage/storage_context.py

```diff
@@ -148,12 +187,25 @@ def to_dict(self) -> dict:
     def from_dict(cls, save_dict: dict) -> "StorageContext":
         """Create a StorageContext from dict."""
         docstore = SimpleDocumentStore.from_dict(save_dict[DOC_STORE_KEY])
-        vector_store = SimpleVectorStore.from_dict(save_dict[VECTOR_STORE_KEY])
         index_store = SimpleIndexStore.from_dict(save_dict[INDEX_STORE_KEY])
         graph_store = SimpleGraphStore.from_dict(save_dict[GRAPH_STORE_KEY])
+
+        vector_stores: Dict[str, VectorStore] = {}
+        for key, vector_store_dict in save_dict[VECTOR_STORE_KEY].items():
+            vector_stores[key] = SimpleVectorStore.from_dict(vector_store_dict)
+
         return cls(
             docstore=docstore,
             index_store=index_store,
-            vector_store=vector_store,
+            vector_stores=vector_stores,
             graph_store=graph_store,
         )
+
+    @property
+    def vector_store(self) -> VectorStore:
```

Comment:
nit should we return vector store dict here instead of default vector store

Reply:
this was acutally needed for backwards compat -- quite a few places used this attribute

Reply:
ic got it